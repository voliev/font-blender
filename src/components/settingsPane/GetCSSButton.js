import React from "react";
import PropTypes from "prop-types";
import Code from "@material-ui/icons/CodeRounded";
import "./styles/GetCSSButton.css";

const GetCSSButton = ({ disabled, getCSSButtonClickHandler }) => (
  <button
    className="GetCSSButton btn-blue-rect"
    disabled={disabled}
    name="getStyles"
    onClick={getCSSButtonClickHandler}
    title="Get CSS"
    value="getStyles"
  >
    <span className="icon">
      <Code fontSize="small" />
    </span>
    Get CSS
  </button>
);

GetCSSButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  getCSSButtonClickHandler: PropTypes.func.isRequired
};

export default GetCSSButton;
