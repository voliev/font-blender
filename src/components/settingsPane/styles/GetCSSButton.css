.GetCSSButton {
  margin: 0 var(--space-md) 0 0;
}

.GetCSSButton .icon {
  display: flex;
  margin-right: var(--space-xs);
}
