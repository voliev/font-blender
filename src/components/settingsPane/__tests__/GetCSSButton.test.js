import React from "react";
import { mount, shallow } from "enzyme";
import GetCSSButton from "../GetCSSButton";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    disabled: false,
    getCSSButtonClickHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<GetCSSButton {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { disabled, getCSSButtonClickHandler, wrapper } = setup(mount);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("getCSSButtonClickHandler")).toBe("function");
  expect(wrapper.prop("getCSSButtonClickHandler")).toBe(
    getCSSButtonClickHandler
  );
});

it("should handle button click", () => {
  const { getCSSButtonClickHandler, wrapper } = setup();

  wrapper.find("button").simulate("click");

  expect(getCSSButtonClickHandler).toHaveBeenCalledTimes(1);
});
