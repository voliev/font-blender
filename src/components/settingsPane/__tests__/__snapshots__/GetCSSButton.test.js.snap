// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<button
  className="GetCSSButton btn-blue-rect"
  disabled={false}
  name="getStyles"
  onClick={[MockFunction]}
  title="Get CSS"
  value="getStyles"
>
  <span
    className="icon"
  >
    <CodeRoundedIcon
      fontSize="small"
    />
  </span>
  Get CSS
</button>
`;
